src/distributions/weighted_index.rs

```diff
@@ -387,6 +379,7 @@ pub trait Weight: Clone {
     /// -   `Result::Err`: Returns an error when `Self` cannot represent the
     ///     result of `self + v` (i.e. overflow). The value of `self` should be
     ///     discarded.
+    #[allow(clippy::result_unit_err)]
```

Comment:
Should we introduce a separate error type as suggested by this lint?

Reply:
Move to a new PR please

Reply:
You mean leave the `allow` for now and introduce new error type in a later PR?

Reply:
No. Given how much needs reformatting, it should be in a dedicated PR (not mixed up with Clippy / other fixes).

Reply:
Doing so will be REALLY annoying to untangle and I would strongly prefer to do both formatting and fixing of Clippy lints in one PR.

Reply:
This attr still shouldn't be here. Let Clippy complain in the CI if necessary.

Reply:
It would fail the CI job. Are you ready to merge a PR with a failing CI job? What is the reason to leave it for a later PR?